cavern/src/test/java/org/opencadc/cavern/nodes/NodeUtilTest.java

```diff
@@ -558,6 +558,34 @@ public void testList() {
                     num++;
                 }
                 Assert.assertEquals("num siblings", i + 1, num);
+
+                // count children for limit == 0 (no children)
+                int upperLimit = 0;
+                parent = nn.getParent();
+                iter = NodeUtil.list(root, parent, null, upperLimit);
+                num = 0;
+                while (iter.hasNext()) {
+                    Node nf = iter.next();
+                    log.info("[testList] found: " + nf.getUri());
+                    num++;
+                }
+                Assert.assertEquals("num siblings", upperLimit + 1, num);
```

Comment:
Hmm, why is one node expected here instead of zero?

Reply:
If I understand the code correctly, num is the number of nodes which includes the parent node as well as the children nodes. So num == 1 accounts for the parent node.

Reply:
NodeUtil.list() simply returns an iterator of the directory, and num starts at 0, so I don't see where the parent is counted...